src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -246,9 +247,60 @@ object TezosTypes {
   //for now we ignore internal results, as it gets funny as sitting naked on a wasps' nest
   final case class ResultMetadata[RESULT](
       operation_result: RESULT,
-      balance_updates: List[OperationMetadata.BalanceUpdate]
+      balance_updates: List[OperationMetadata.BalanceUpdate],
+      internal_operation_results: Option[List[InternalOperationResults.InternalOperationResult]] = None
   )
 
+// Internal operations result definitions
+  object InternalOperationResults {
+
+    sealed trait InternalOperationResult {
```

Comment:
Extending Product with Serializable should generally be considered good practice

Reply:
ok, added